Chart Node.js API support over time, remove unenv

irvinebroque opened the pull request:
- Removes any reliance on unenv, eliminates any confusion about whether we are counting mocked APIs or not
- Updates the underlying data (last ran in September)

Adding `http`, `https`, `fs`, and a bunch of other misc things in September/October drove overall % compatibility way up:

<img width="1035" height="157" alt="Screenshot 2026-01-03 at 5 15 49 PM" src="https://github.com/user-attachments/assets/b650ea16-7340-47c9-a172-fcd1e1e62cad" />
<img width="845" height="561" alt="Screenshot 2026-01-03 at 5 15 58 PM" src="https://github.com/user-attachments/assets/026ade67-1bb7-43a2-b34b-1f1ae86ccbbb" />

irvinebroque committed:
Vibe code chart over time between meetings

anonrig commented on generate-historical-data.mjs:
```diff
@@ -0,0 +1,416 @@
+    const moduleName = keyPath[0];
+    const moduleInfo = target[moduleName];
+
+    // mock module can be identified as a module that has only one key - the default,
```
This is not correct. Are you sure?

irvinebroque replied:
I did say vibe coded :)

lemme go through this, am sure not quite right

irvinebroque replied:
well wait, this is from here

https://github.com/cloudflare/workers-nodejs-compat-matrix/blob/main/generate-table-data.mjs#L70

irvinebroque committed:
Rip out unenv, regenerate data and chart

irvinebroque committed:
Update

irvinebroque committed:
Remove references to mock data

irvinebroque committed:
Update baseline.json with Node.js 22.x API additions

Add newly available APIs from recent Node.js versions:

- Explicit Resource Management: AsyncDisposableStack, DisposableStack, SuppressedError
- Float16 support: Float16Array, Math.f16round
- Web APIs: CloseEvent, URLPattern, navigator.locks (Web Locks API)
- Intl.DurationFormat for duration formatting
- Atomics.pause for spinlock optimization
- Crypto ML-KEM post-quantum methods (encapsulate/decapsulate)
- crypto.argon2/argon2Sync for Argon2 password hashing
- process.threadCpuUsage for worker thread CPU metrics
- http.Agent.agentKeepAliveTimeoutBuffer
- assert.Assert class

Also reflects platform-specific constant differences (macOS vs Linux).
